src/libsync/syncengine.cpp

```diff
@@ -889,7 +916,7 @@ void SyncEngine::slotDiscoveryJobFinished(int discoveryResult)
     _journal->commit("post treewalk");
 
     _propagator = QSharedPointer<OwncloudPropagator>(
-        new OwncloudPropagator (_account, _localPath, _remoteUrl, _remotePath, _journal));
+        new OwncloudPropagator (_account, _localPath, _remoteUrl.path(), _remotePath, _journal));
```

Comment:
It is the same, I just moved the .path() from Folder and OwnCloudCmd in here so that I could move the weird scheme logic above in here instead of outside. What we gain is that remoteUrl is an actual url (else it should be named remotePath, but we already have a remotePath, so anyway).